jwst/outlier_detection/outlier_detection_step.py

```diff
@@ -163,13 +165,20 @@ def process(self, input_data):
 
             # Set up outlier detection, then do detection
             step = detection_step(self.input_models, reffiles=reffiles, **pars)
-            step.do_detection()
+            try:
+                step.do_detection()
+            except RuntimeError as exception:
+                self.log.error(f'{exception}')
+                state = 'SKIP'
```

Comment:
The issue description claims that the desired behavior is to "shutdown cleanly with a clear error message".  Skipping with a log message seems more hazardous, since a user might not notice that the step had been skipped entirely.

Reply:
Either way. Part of the point of stpipe is to do things like this, and there are other cases. So, the user should expect such things. However, the community "we" have not been diligent on this point, so other reviewers chime in for further opinion.

The commentary may be a bit optimistic though. :-)

Reply:
@hbushouse A statement you just made in the integration meeting probably defines what should happen: If science processing cannot continue, the pipeline will raise an exception. Paraphrased anyways. Seems like this situation would count as "processing cannot continue". As such, the exception will not be caught, though sufficient error messages should make the situation clear.